### ISSUE TITLE
Proposal: Warn/error in transaction pooling mode when unsupported features are used

### ISSUE THREAD

**Issue by JelteF** [pgbouncer/pgbouncer#976]
In transaction pooling there are quite a few Postgres features that are unsupported. Most importantly:
- SQL level prepared statements, i.e. PREPARE/EXECUTE/DEALLOCATE
- LISTEN (NOTIFY is fine)

#972 is using a clever approach to detect that (currently unsupported) DEALLOCATE ALL/DISCARD ALL have been used, without actually needing any query parsing logic. We could use the same approach to detect PREPARE/EXECUTE/DEALLOCATE/LISTEN queries, and warn/error/close the connection. I feel that being loud about bad usage probably better for the end user than silently accepting things that will almost certainly break.

**Comment by kelvich** [comment]
Just faced an interesting foot gun:
* run `pg_dump` on your database though pgbouncer
* `pg_dump` will issue `SET search_path ...` that will be ingested in one of shared connection to postgres
* that will "poison" that connection and some random database queries will fail to find a table

**Comment by JelteF** [comment]
Yeah this is a known issue, that is tracked here: https://github.com/pgbouncer/pgbouncer/issues/452

I responded to that one just now with the current state of possible workarounds.
